### PR TITLE
Feature/optional dependencies

### PR DESCRIPTION
**Pull request by vonabarak**
- Dependencies for asyncbox package are now optional and depends on selected plugins.
- Added setting `DEFAULT_ROUTER` for ability to change BotX API requests handlers logic
- In SQLAlchemy plugin attribute `session` replaced with method `make_session()`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -36,11 +37,30 @@ pytest-clarity = "~0.3.0-alpha.0"
 pytest-timeout = "~1.3.4"
 asgi-lifespan = "~1.0.1"
 molten = "~1.0.2"
-sqlalchemy-stubs = "~0.4"
+
+sqlalchemy2-stubs = "^0.0.2-alpha.18"
+
+# extras packages are duplicated to dev-dependencies as they need for unit-tests
```
Comment (lxmnk) [inline pyproject.toml:43]: Maybe use [pytest.importorskip](https://docs.pytest.org/en/latest/how-to/skipping.html#skipping-on-a-missing-import-dependency)?

File: asyncbox/plugins/sqlalchemy/plugin.py [hunk 2]
```diff
@@ -20,30 +20,30 @@ class SQLAlchemyPlugin(BasePlugin):
...
+            raise RuntimeError("Plugin not yet initialized!")
+        return self._make_session
 
     async def healthcheck(self) -> HealtCheckData:
         """Healthcheck function."""
-        try:
-            async with self.session.begin():
-                rows = await self.session.execute("select version()")
+        try:  # noqa: WPS229
+            session = self.make_session()
```
Comment (lxmnk) [inline asyncbox/plugins/sqlalchemy/plugin.py:44]: You can use:
```async with self.make_session() as session, session.begin():```

[link to docs](https://docs.sqlalchemy.org/en/14/orm/session_basics.html?highlight=session#framing-out-a-begin-commit-rollback-block)

**Review by YunusovSamat — APPROVED** [review]

**Review by Nimond — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (vonabarak) [reply pyproject.toml:43]: In this case some tests will  be just skipped in CI/CD.

File: asyncbox/plugins/sqlalchemy/plugin.py (hunk 2 above)
Comment (vonabarak) [reply asyncbox/plugins/sqlalchemy/plugin.py]: done

**Review by lxmnk — APPROVED** [review]

**Pull request merged**